### PR TITLE
chore: migrate @fluentui/react-theme

### PR DESCRIPTION
**Pull request by layershifter**
#### Pull request checklist

- [x] Addresses an existing issue: related to #20597
- [x] Include a change request file using `$ yarn change`

#### Description of changes

- Ran generator

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/59b3bebb1944aa35f8e8754beec16dcc95f1d5fb'>59b3bebb1944aa35f8e8754beec16dcc95f1d5fb</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-theme/tsconfig.json [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
+  "references": [
+    {
+      "path": "./tsconfig.lib.json"
+    },
+    {
+      "path": "./tsconfig.spec.json"
+    },
+    {
+      "path": "./.storybook/tsconfig.json"
+    }
```
Comment (ling1726) [inline packages/react-theme/tsconfig.json:23]:
```suggestion
    }
```

**Review by ling1726 — APPROVED** [review]

File: packages/react-theme/tsconfig.json (hunk 1 above)
Comment (ling1726) [reply packages/react-theme/tsconfig.json:23]: oh wait, you didn't remove storybook... please update the PR description

Comment (layershifter) [reply packages/react-theme/tsconfig.json:23]: This package contains stories, so we should keep this config

**Pull request merged**